src/internal.h

```diff
@@ -652,10 +653,12 @@ struct ThreadState {
     ThreadState() = default;
     ThreadState(const ThreadState &other) = default;
 
-    virtual Task *launch(Kernel kernel, uint32_t size,
-                         std::vector<void *> *kernel_params,
-                         uint32_t kernel_param_count,
-                         const uint8_t *kernel_params_global) = 0;
+    /// Inserts a barrier task
```

Comment:
Expand a little bit on what this means, whether all backends are supported or not, etc.